### PR TITLE
Update Helper.js

### PR DESCRIPTION
**Pull request by lipflip**
Fix for wrong 7bit decode, described here: https://github.com/jackkum/node-pdu/issues/20#issuecomment-478877023

### CODE REVIEW

**Comment by Apollon77** [comment]
Proofed that fix by testing. We also mojey patched it in serialport-gsm (https://github.com/zabsalahid/serialport-gsm/pull/115/files)

File: PDU/Helper.js [hunk 1]
```diff
@@ -129,12 +129,35 @@ Helper.decode7Bit = function(text, inLen, alignBits)
...
+          }
+          if (c < 0x80) {
+              ret.push(c);
+          } else if (c < 0x800) {
+              ret.push(0xC0 | (c >> 6), 0x80 | (c & 0x3F));
+          } else if (
+              ((c & 0xFC00) == 0xD800) && (digit + 1) < Helper.EXTENDED_TABLE.length &&
+              ((Helper.EXTENDED_TABLE.charCodeAt(digit + 1) & 0xFC00) == 0xDC00)) {
+              // Surrogate Pair
+              c = 0x10000 + ((c & 0x03FF) << 10) + (str.charCodeAt(++digit) & 0x03FF);
```
Comment (KillerJulian) [inline PDU/Helper.js:147]: Where does the variable `str` come from? 🤔

**Review by KillerJulian — COMMENTED** [review]
Thank you for these changes. I will include them in my refactor #25

File: PDU/Helper.js (hunk 1 above)
Comment (Apollon77) [reply PDU/Helper.js:147]: uups... in fact you are right ... interestingly this never had an issue in my tests :-)

Comment (Apollon77) [reply PDU/Helper.js:147]: @lipflip how it should be correctly?

Comment (Apollon77) [reply PDU/Helper.js:147]: I assume str needs to be "Helper.EXTENDED_TABLE", makes that sense?

Comment (KillerJulian) [reply PDU/Helper.js:147]: I guess

Comment (Apollon77) [reply PDU/Helper.js:147]: Then two guesses is best :-)  I also have no idea in which cases this fallback would get active ... ideally we have a test for that, but thats not that simple :-)

**Comment by KillerJulian** [comment]
Thanks for contributing!
Your changes were introduced with the pull request https://github.com/jackkum/node-pdu/pull/25 .

Yours sincerely
Julian

**Pull request closed without merging**